### DISCUSSION TITLE
Is it possible to publish a reactable table from Quarto to Confluence?

### DISCUSSION THREAD

**Discussion by catterbury-cai** [quarto-dev/quarto-cli#5160] [Q&A] [answered]
I have a Quarto document originally published in HTML format. I use the `reactable` package to display data returned from queries to my company's databases.
I attempted to publish to Confluence with the most recent Quarto update and got the following error:
```{terminal}
Error: Functions that produce HTML output found in document targeting publish output.
Please change the output type of this document to HTML. Alternatively, you can allow
HTML output in non-HTML formats by adding this option to the YAML front-matter of
your quarto file:

  prefer-html: true

Note however that the HTML output will not be visible in non-HTML formats.

Execution halted
```
I traced this to `reactable` as it was able to publish when excluding use of that package. As an alternative, I tried using `kableExtra` and found the same error as soon as I introduced elements that leverage that package's capabilities. I know that not all HTML and Quarto elements are supported, but my current options appear to be an extremely basic table that is not really usable with the results of my queries, or an error.
When I run the `quarto preview` command, everything is fine, but actually publishing to Confluence using `quarto publish confluence` generates the above error whenever any HTML element is introduced. When I exclude them, it writes out to an xml file.
Is this expected behavior? Is there a workaround? Thanks!

**Comment by cscheid** [comment, marked as answer]
The confluence format does not support full HTML, so unfortunately there's not much we can do in this case.

**Reply by catterbury-cai** [reply]
Okay. Thanks!

**Comment by ghost** [comment]
This was [also asked here](https://github.com/quarto-dev/quarto-cli/discussions/2003#discussioncomment-5257198).

We can explore many different additions as we move forward. One very important factor is the [limitations of CSF and what they will allow us to do](https://confluence.atlassian.com/doc/confluence-storage-format-790796544.html).

**Reply by cderv** [reply]
A special note based on R Markdown experience. Just sharing thoughts on this. 

Here we are talking about tables, and especially paginated table, so maybe this does not apply, but for document using **knitr**, any HTML widgets (like reactibable output) will be screenshot to an image when **webshot** or **webshot2** R package are installed. 
This allow to insert any HTML only content into other format that don't support them. 

This logic could be extended to confluence format maybe so that complex output like HTML widget could still be included in a confluence page but as image. This would be one step probably. We could think of something more clever according to what Confluence allows. 

Anyhow, I don't think Quarto handles this HTML-object conversion to something else compatible to a format as well as R Markdown us doing it (to avoid errors when rendering to multiple format)
